examples/gno.land/p/demo/grc/grc721/grc721_metadata_test.gno

```diff
@@ -0,0 +1,134 @@
+package grc721
+
+import (
+	"std"
+	"testing"
+
+	"gno.land/p/demo/testutils"
+	"gno.land/p/demo/users"
+)
+
+func TestSetMetadata(t *testing.T) {
+	// Create a new dummy NFT with metadata
+	dummy := NewNFTWithMetadata(dummyNFTName, dummyNFTSymbol)
+	if dummy == nil {
+		t.Errorf("should not be nil")
+	}
+
+	// Define addresses for testing purposes
+	addr1 := std.Address("g1var589z07ppjsjd24ukm4uguzwdt0tw7g47cgm")
+	addr2 := std.Address("g1us8428u2a5satrlxzagqqa5m6vmuze025anjlj")
+
+	// Define metadata attributes
+	name := "test"
+	description := "test"
+	image := "test"
+	imageData := "test"
+	externalURL := "test"
+	attributes := []Trait{}
+	backgroundColor := "test"
+	animationURL := "test"
+	youtubeURL := "test"
+
+	// Set the original caller to addr1
+	std.TestSetOrigCaller(std.Address(addr1)) // addr1
+
+	// Mint a new token for addr1
+	dummy.mint(addr1, TokenID("1"))
+
+	// Set metadata for token 1
+	_, derr := dummy.SetTokenMetadata(TokenID("1"), Metadata{
+		Name:           name,
+		Description:    description,
+		Image:          image,
+		ImageData:      imageData,
+		ExternalURL:    externalURL,
+		Attributes:     attributes,
+		BackgroundColor: backgroundColor,
+		AnimationURL:   animationURL,
+		YoutubeURL:     youtubeURL,
+	})
+
+	// Check if there was an error setting metadata
+	if derr != nil {
+		t.Errorf("Should not result in error : %s", derr.Error())
+	}
+
+	// Test case: Invalid token ID
+	_, err := dummy.SetTokenMetadata(TokenID("3"), Metadata{
+		Name:           name,
+		Description:    description,
+		Image:          image,
+		ImageData:      imageData,
+		ExternalURL:    externalURL,
+		Attributes:     attributes,
+		BackgroundColor: backgroundColor,
+		AnimationURL:   animationURL,
+		YoutubeURL:     youtubeURL,
+	})
+
+	// Check if the error returned matches the expected error
+	if err != ErrInvalidTokenId {
+		t.Errorf("Expected error %s, got %s", ErrInvalidTokenId, err)
+	}
+
+	// Set the original caller to addr2
+	std.TestSetOrigCaller(std.Address(addr2)) // addr2
+
+	// Try to set metadata for token 1 from addr2 (should fail)
+	_, cerr := dummy.SetTokenMetadata(TokenID("1"), Metadata{
+		Name:           name,
+		Description:    description,
+		Image:          image,
+		ImageData:      imageData,
+		ExternalURL:    externalURL,
+		Attributes:     attributes,
+		BackgroundColor: backgroundColor,
+		AnimationURL:   animationURL,
+		YoutubeURL:     youtubeURL,
+	})
+
+	// Check if the error returned matches the expected error
+	if cerr != ErrCallerIsNotOwner {
+		t.Errorf("Expected error %s, got %s", ErrCallerIsNotOwner, cerr)
+	}
+
+	// Set the original caller back to addr1
+	std.TestSetOrigCaller(std.Address(addr1)) // addr1
+
+	// Retrieve metadata for token 1
+	dummyMetadata, err := dummy.TokenMetadata(TokenID("1"))
+	if err != nil {
+		t.Errorf("Metadata error: %s", err.Error())
+	} else {
```

Comment:
Just like in the other test file, this else doesn't do anything, you can remove it and un-indent the if statements

Reply:
https://github.com/linhpn99/gno/commit/e1e10e8053efea1ee747bbcba38ec24a75d724cf